### DISCUSSION TITLE
Expose `more` property on `Pagy::Keyset`

### DISCUSSION THREAD

**Discussion by olivier-thatch** [ddnexus/pagy#734] [Feature Requests]
Pagy's keyset implementation queries one more record than actually requested, then checks the number of records returned by the database to determine whether the end of the set has been reached or not: https://github.com/ddnexus/pagy/blob/c5a7b92c2bed7786ec422a522aac5ec04c49bf0b/gem/lib/pagy/keyset.rb#L64-L65

Currently, `@more` is only used internally and is not exposed to consumers of the API. I think it should: for instance, it would be useful to expose this property in a public API, so that API clients can know whether they've reached the end of the set or not without having to send an additional request.

As an example of real-word use, Stripe's API uses cursor-based pagination and its paginated list object includes an [`has_more`](https://docs.stripe.com/api/pagination#pagination-has_more) property.

If you think this would be a useful addition to Pagy, I'm happy to submit a PR. It should literally be as simple as adding an `attr_reader` :)

**Comment by ddnexus** [comment]
> If you think this would be a useful addition to Pagy, I'm happy to submit a PR.

That's very welcome! Thank you! I am just not so sure we need it.

> it would be useful to expose this property in a public API, so that API clients can know whether they've reached the end of the set or not without having to send an additional request.

Isn't that already possible with `@pagy.next.nil?` ?
